burr/core/application.py

```diff
@@ -836,18 +848,10 @@ def streaming_response(state: State, prompt: str) -> Generator[dict, None, Tuple
             # For context, this is specifically for the case in which you want to have
             # multiple terminal points with a unified API, where some are streaming, and some are not.
             if next_action.name in halt_before and next_action.name not in halt_after:
-                self._adapter_set.call_all_lifecycle_hooks_sync(
-                    "post_run_step",
```

Comment:
I assume this is being called somewhere else, and this was being called twice?

Reply:
Yep, we delegate to `run`